File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        
        # 1, 2, 3, 4
        # target = 3
        # 1 and 4
        # 1 + 4 = 5
        # too big. 
        # 1 + 3 = 4
        # too big. 
        # 1 + 2 = 3
        while left < right:
            lv = numbers[left]
            rv = numbers[right]
            twoSum = lv + rv

            if twoSum == target:
                return [left + 1, right + 1]
            elif twoSum > target:
                right -= 1
            else:
                left += 1
        return []

File: Data Structures & Algorithms/find-first-and-last-position-of-element-in-sorted-array/submission-3.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        
        int start = binarySearch(nums, target, nums.length);
        if (start == nums.length || nums[start] != target) {
            return new int[]{-1, -1};
        }


        return new int[]{start, binarySearch(nums, target + 1, nums.length) - 1};
        
    }

    public int binarySearch(int[] nums, int target, int n) {
        int left = 0;
        int right = n;

        while (left < right) {
            final int m = left + (right - left) / 2;
            
            if (nums[m] >= target) right = m;
            else left = m + 1;
        }

        return left;
    }
}